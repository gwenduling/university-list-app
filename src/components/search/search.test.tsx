import Search from "./search";

import { shallow } from "enzyme";
import { DebounceInput } from "react-debounce-input";

describe("Search Component", () => {
  it("should match the snapshot", () => {
    const wrapper = shallow(
      <Search placeholder="" label="" value="" onValueChanged={() => {}} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should trigger callback", () => {
    const mockCallBack = jest.fn();
    const wrapper = shallow(
      <Search
        placeholder=""
        label=""
        value=""
        onValueChanged={() => mockCallBack()}
      />
    );

    wrapper.find(DebounceInput).simulate("change", { target: { value: "" } });
    expect(mockCallBack).toHaveBeenCalledTimes(1);
  });
});
