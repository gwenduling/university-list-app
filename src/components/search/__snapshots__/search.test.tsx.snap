// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component should match the snapshot 1`] = `
<div
  className="search-input"
>
  <label
    className="label"
  />
  <DebounceInput
    debounceTimeout={200}
    element="input"
    forceNotifyByEnter={true}
    forceNotifyOnBlur={true}
    minLength={0}
    onChange={[Function]}
    placeholder=""
    type="text"
    value=""
  />
</div>
`;
